src/System.Linq/src/System/Linq/Count.cs

```diff
@@ -15,6 +15,8 @@ public static int Count<TSource>(this IEnumerable<TSource> source)
             if (source == null) throw Error.ArgumentNull("source");
             ICollection<TSource> collectionoft = source as ICollection<TSource>;
             if (collectionoft != null) return collectionoft.Count;
+            IIListProvider<TSource> listProv = source as IIListProvider<TSource>;
+            if (listProv != null) return listProv.GetCount(false);
```

Comment:
Sure. Of the three places this was called that weren't passing through that value, there was this and two cases where `ListPartition` called it on itself. I've made those two into a call to a private `Count` property, to make it clearer that the flag isn't relevant to that use.

That leaves this as the sole actual use of it, but I have plans for it :smile: